### DISCUSSION TITLE
Loading a recursive graph's relationships

### DISCUSSION THREAD

**Discussion by Lacrymology** [sqlalchemy/sqlalchemy#11800] [Usage Questions]
Hello. I've got a nested recursive structure: https://gist.github.com/Lacrymology/d7d49fb40c63f5c4140ce40607728954

A `PackingConfig` has `Pallets`, `Pallets` have `Packs`, `Packs` have other `Packs`, `CasePacks`, or `LineItems`. 

The nice part is that every object regardless of the position in the graph has a reference to the root object. I would like to build the graph with a single query (basically, select from the packing config left join with pallets packs and case packs filter by a particular packing config id), but as far as I can see`contains_eager` doesn't seem to be smart enough to assign everything correctly. Maybe I'm wrong, or using it wrong, or maybe someone can tell me some trick to do this? I'm okay with traversing each of the lists on the packing config and populating each object's children, but it's pretty critical that I don't trigger any extra DB roundtrips. Also I think this should be possible :thinking:

**Comment by CaselIT** [comment]
Hi,

Before resorting to manually writing the query then using containe eager, have you tried using the eager loaders available in sqlalchemy?

Joinedload uses joins and can use w single query, but note that depending on the cardinality it may be more efficient to switch to selecinload after a couple of levels

**Reply by Lacrymology** [reply]
@CaselIT the problem with that is that I need to write however many levels of depth I want to load, right? My type has arbitrary depth. And it'll throw several queries when with just one I can grab all the necessary data.

Maybe you can provide a simple example? You've got all the relevant data for the models in that gist
